nginx/nginx.tmpl

```diff
@@ -102,9 +104,21 @@ http {
 
 {{- range $entry := .Servers }}
     # ingress: {{ $entry.Name }}
+  {{- range $IngressPortConf := $IngressPorts }}
     server {
-        listen {{ $port }}{{ if $proxyprotocol }} proxy_protocol{{ end }};
+        listen {{ $IngressPortConf.Port }}{{ if $proxyprotocol }} proxy_protocol{{ end }}{{- if eq $IngressPortConf.Name "https" }} ssl {{ end }};
         server_name {{ $entry.ServerName }};
+{{- if eq $IngressPortConf.Name "https" }}
+        ssl_certificate     {{ $DefaultSslPath }}.crt;
+        ssl_certificate_key  {{ $DefaultSslPath }}.key;
+        # secure settings (A+ at SSL Labs ssltest at time of writing)
+        # see https://wiki.mozilla.org/Security/Server_Side_TLS#Nginx
+        ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+        ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:DHE-RSA-CAMELLIA256-SHA:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-SEED-SHA:DHE-RSA-CAMELLIA128-SHA:HIGH:!aNULL:!eNULL:!LOW:!3DES:!MD5:!EXP:!PSK:!SRP:!DSS';
+        ssl_prefer_server_ciphers on;
+        ssl_session_cache shared:SSL:60m;
+        ssl_session_timeout 5m;
```

Comment:
These values don't seem to match the mozilla config. We also want     `ssl_session_tickets off;` so we get forward security.